protocol/chat.go

```diff
@@ -391,6 +398,14 @@ func (c *Chat) UpdateFromMessage(message *common.Message, timesource common.Time
 	return nil
 }
 
+func (c *Chat) ShouldIncreaseMessageCount(message *common.Message) bool {
+	return c.LastMessage == nil ||
+		c.LastMessage.IsSystemMessage() ||
+		!c.LastMessage.New ||
+		c.LastMessage.Seen ||
```

Comment:
This line was missing in the original PR. Without it, if you saw the last message and went to another chat, the marker wouldn't increase.